lib/parser.js

```diff
@@ -0,0 +1,150 @@
+'use strict'
+
+const Parser = require('htmlparser2/lib/Parser')
+
+const DIRECTIVES = [
+  { name: '!doctype', start: '<', end: '>' },
+  { name: '?php', start: '<', end: '>' }
+]
+
+const SELF_CLOSING = [
+  'area',
+  'base',
+  'br',
+  'col',
+  'embed',
+  'hr',
+  'img',
+  'input',
+  'keygen',
+  'link',
+  'meta',
+  'param',
+  'source',
+  'track',
+  'wbr',
+  // Custom (PostHTML)
+  'import',
+  'include',
+  'component'
+]
+
+/**
+ * Parse HTML to PostHTML Tree
+ *
+ * @method parser
+ *
+ * @param  {String} html
+ * @param  {Object} options
+ *
+ * @return {Array}  result
+ */
+function parser (html, options) {
+  const buffer = []
+
+  buffer.last = function () {
+    return this[this.length - 1]
+  }
+
+  const result = []
+
+  const selfClosing = SELF_CLOSING.concat(options.tags || [])
```

Comment:
Relevant

Reply:
no, you need to add a parameter `selfClosing` in `tree`, and check if the `tag` has a closing slash and does not have `content`
```html
<customTag />
```

```js
tag: 'customTag',
selfClosing: true
```

Reply:
I'm not sure I fully understand, are you suggesting to add a `selfClosing` prop to the `node` instead ?

```js
{
  tag: 'import'
  selfClosing: true // <=
  attrs: { ... }
}
```

?

The `SELF_CLOSING` list of tags might be unneeded here, I definitely need to revisit this once more, maybe I mixed things up here 😛 and we only need to check the node shape
 to not have any  `content`, like you suggested